replica/apply_mutation.hh

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <seastar/core/sharded.hh>
+
+#include "schema.hh"
+
+using namespace seastar;
+
+class mutation;
+class frozen_mutation;
+
+namespace replica {
+
+class apply_mutation {
```

Comment:
I don't understand the name. It looks like this is std::variant<mutation, frozen_mutation>, but why the name apply_mutation? That's also a verb, not a noun.

Reply:
It resembles a `std::variant<mutation, frozen_mutation>` but it can also keep a pointer to an external mutation or frozen_mutation, if constructed with one.

Since a mutation to be applied, how about `applied_mutation`?
Or would you rather it'd just describe what it holds, e.g. `mutation_or_frozen_mutation`?

Reply:
The last one is the least confusing.